agnpy/time_evolution/tests/test_time_evolution.py

```diff
@@ -312,9 +311,174 @@ def test_simulation_trending_to_delta_function(self):
+            offset += (nrows+1)
+
+        # prepare objects
+        R_b = (100 * c * u.s).to(u.cm)
+        V_b = 4 / 3 * np.pi * R_b ** 3
+        distance = Distance(z=0.01)
+        delta_D = 1.001
+        Gamma = 1.001
+        B = 10.3 * u.G
+        tacc = 10 * u.s
+        tesc = 10 * u.s
+        blob = Blob(R_b, distance.z, delta_D, Gamma, B, n_e=EmptyDistribution(gamma_min=10, gamma_max=1e3))
+        synch = Synchrotron(blob)
+        injection_dist = PowerLaw.from_total_energy(100000 * u.GeV, V_b, gamma_min=10, gamma_max=20, p=2, mass=m_e)
+        def injection_function_zone1(args):
+            return injection_dist(args.gamma) / u.s
+
+        def escape_fn(args):
+            return -1 * np.ones(args.densities.shape) / tesc
+
+        def injection_function_zone2(args):
+            return -1 * escape_fn(args) * args.densities * args.density_subgroups[0]
+
+        # run simulations over different time scales
+        times = [0, 11, 31, 101, 301]
+        number_of_bins = 11
+        density_groups = np.array([np.ones(number_of_bins, dtype=float), np.zeros(number_of_bins, dtype=float)])
+        gamma_array = np.logspace(np.log10(10), np.log10(20), number_of_bins)
+        for i in range(len(times)-1):
+            single_iteration_time = (times[i + 1] - times[i]) * u.s
+            _, gamma_array, densities, density_groups, energy_change_rates, _, _ = (TimeEvolution(
+                blob,
+                single_iteration_time,
+                energy_change_functions={"Synch": synchrotron_loss(synch), "Acc": fermi_acceleration(tacc)},
+                injection_functions_abs={"Zone1-inj": injection_function_zone1, "Zone2-inj": injection_function_zone2},
+                injection_functions_rel={"Zone1-esc": escape_fn},
+                subgroups=[["Synch", "Acc", "Zone1-inj", "Zone1-esc"],
+                           ["Synch", "Zone2-inj"]],
+                subgroups_initial_density=density_groups,
+                initial_gamma_array=gamma_array,
+                gamma_bounds=(1e1, 1e9),
+                method="euler",
+                step_duration=0.2 * u.s,
+                max_bin_creep_from_bounds=0.005,
+            ).evaluate())
+
+            distribution_zone1 = InterpolatedDistribution(gamma_array, densities * density_groups[0])
+            distribution_zone2 = InterpolatedDistribution(gamma_array, densities * density_groups[1])
+            ref = ref_data[i]
+            eed_x = ref.eed_x.values
+            eed_y_zone1 = to_total_energy_gev_sqr(eed_x, distribution_zone1(eed_x), V_b)
+            eed_y_zone2 = to_total_energy_gev_sqr(eed_x, distribution_zone2(eed_x), V_b)
+
+
+            df = pd.DataFrame({
+                "eed_x": ref.eed_x.values,
+                "eed_y_zone1": eed_y_zone1.value,
+                "eed_y_zone2": eed_y_zone2.value
+            })
+            my_data.append(df)
+
+            ignore_pos = [8,13,29] # positions where sharp density changes occur
```

Comment:
strange to have it in 3 points, or are those problematic points show up for different times?